### PR TITLE
fix: conversation messages and event order

### PR DESCRIPTION
**Pull request by araujobarret**
The type of this PR is: **Bugfix**

[NX-3102]

### Description

When the conversation starts from Make Offer we don't see events being rendered in the conversation, like the message `You sent an Offer of $XXX`.
Another fix included is to show the conversation CTA to interact with counter offers given conversations starting from Make Offer.
There are some open questions like messages and events that are not grouped at the same time to render but are not related to the scope of this PR.

Here the first 2 conversations started from Make Offer flow, the last one started from Inquiry.
https://user-images.githubusercontent.com/15792853/157627846-625c227b-e669-4ed9-95e2-f41b5633fb1e.mov

Before (Conversation started from MO given counter-offer)
image

After (Conversation started from MO given counter-offer)
image

cc @artsy/negotiate-devs 

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/main/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/main/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [ ] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- fix events/ctas not rendered in conversation when not from inquiry - araujobarret

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

[NX-3102]: https://artsyproduct.atlassian.net/browse/NX-3102?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (fix events not rendered in conversation when not from inquiry - araujobarret)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 7d3c6b32cf9e2997429fc368b810ae6a50fd187a
</p>

File: src/app/Scenes/Inbox/Components/Conversations/ConversationCTA.tsx [hunk 1]
```diff
@@ -15,21 +15,26 @@ interface Props {
...
 
-  const artwork = liveArtwork?.__typename === "Artwork" ? liveArtwork : null
-  const artworkID = artwork?.internalID
-  const isOfferableFromInquiry = artwork?.isOfferableFromInquiry
+  if (liveArtwork?.__typename !== "Artwork") {
+    return null
+  }
+
+  const artworkID = liveArtwork?.internalID
+  const isOfferableFromInquiry = liveArtwork?.isOfferableFromInquiry
```
Comment (araujobarret) [inline src/app/Scenes/Inbox/Components/Conversations/ConversationCTA.tsx:23]: nothing changed here, just improving the checks

File: src/app/Scenes/Inbox/Components/Conversations/ConversationCTA.tsx [hunk 2]
```diff
@@ -15,21 +15,26 @@ interface Props {
...
 
-  if (inquiryCheckoutEnabled && isOfferableFromInquiry) {
+  if (inquiryCheckoutEnabled) {
     // artworkID is guaranteed to be present if `isOfferableFromInquiry` was present.
     const conversationID = conversation.conversationID!
 
     const activeOrder = extractNodes(conversation.activeOrders)[0]
     if (!activeOrder) {
-      CTA = <OpenInquiryModalButton artworkID={artworkID!} conversationID={conversationID} />
+      if (isOfferableFromInquiry) {
```
Comment (araujobarret) [inline src/app/Scenes/Inbox/Components/Conversations/ConversationCTA.tsx:35]: now we do like Force, we want to enable counter offer and other interactions given when it's not `isOfferableFromInquiry`

File: src/app/Scenes/Inbox/Components/Conversations/Messages.tsx [hunk 3]
```diff
@@ -81,11 +81,7 @@ export const Messages: React.FC<Props> = forwardRef((props, ref) => {
     const sortedMessages = sortBy([...orderEventsWithoutFailedPayment, ...allMessages], (message) =>
       DateTime.fromISO(message.createdAt!)
     )
-
-    // Drop all order events until the first message appears (this assumes the conversation begins with a message)
-    const sortedWithCutoff = dropWhile(sortedMessages, (item) => item.__typename !== "Message")
-
-    const groupedMessages = groupConversationItems(sortedWithCutoff)
+    const groupedMessages = groupConversationItems(sortedMessages)
```
Comment (araujobarret) [inline src/app/Scenes/Inbox/Components/Conversations/Messages.tsx:84]: now we do expect first messages of different types and we render them, don't drop

**Review by lidimayra — APPROVED** [review]
👏 👏 👏 

Not only because it was a super-fast solution, but also: great description and comments!! 💜

**Pull request merged**